### PR TITLE
refactor/oomagent-channelJoin

### LINKED ISSUE #1128: reactor oomagent-channelJoin with cancel context

### PR DESCRIPTION
**Pull request by lianxmfor**
refactor oomagent-channelJoin with cancel context.

close #1128

### CODE REVIEW

File: oomagent/test/test_channel_join.sh [hunk 1]
```diff
@@ -76,3 +76,37 @@ EOF
...
+  Code: Internal
+  Message: invalid full feature name: 'invalid_feature_name'
+EOF
+)
+  assert_eq "$case" "$expected" "$actual"
+}
+
+main() {
+  case1
+  case2
```
Comment (jinghancc) [inline oomagent/test/test_channel_join.sh:109]: suggestion: could rename `case1` as `successful_cases`, `case2` as `failed_cases`

Comment (lianxmfor) [reply oomagent/test/test_channel_join.sh:109]: sounds good!

**Review by YikSanChan — APPROVED** [review]
lgtm

File: oomagent/test/test_channel_join.sh (hunk 1 above)
Comment (lianxmfor) [reply oomagent/test/test_channel_join.sh]: done

**Pull request merged**